tests/functions_tests/test_mean_squared_error.py

```diff
@@ -0,0 +1,167 @@
+    loss_value = cuda.to_cpu(loss.data)
+    assert loss.dtype == numpy.float32
+    assert loss_value.shape == ()
+
+    loss_expect = numpy.zeros(())
+    x0_data = cuda.to_cpu(x0_data)
+    x1_data = cuda.to_cpu(x1_data)
+    for i in numpy.ndindex(x0_data.shape):
+        loss_expect += ((x0_data[i] - x1_data[i]) ** 2)
+    loss_expect /= x0_data.size
+    assert numpy.allclose(loss_value, loss_expect)
+
+
+def check_forward_ignore_nan(inputs):
+    x0_data, _, x2_data = inputs
+    x0 = chainer.Variable(x0_data)
+    x2 = chainer.Variable(x2_data)
+    loss = chainer_chemistry.functions.mean_squared_error(x0, x2,
+                                                          ignore_nan=True)
+    loss_value = cuda.to_cpu(loss.data)
+    assert loss.dtype == numpy.float32
+    assert loss_value.shape == ()
+
+    loss_expect = numpy.zeros(())
+    x0_data = cuda.to_cpu(x0_data)
+    x2_data = cuda.to_cpu(x2_data)
+    nan_mask = numpy.invert(numpy.isnan(x2_data)).astype(x2_data.dtype)
+    for i in numpy.ndindex(x0_data.shape):
+        loss_expect += ((x0_data[i] - numpy.nan_to_num(x2_data[i])) ** 2
+                        * nan_mask[i])
+    loss_expect /= x0_data.size
+    assert numpy.allclose(loss_value, loss_expect)
+
+
+def check_forward_ignore_nan_with_nonnan_value(inputs):
+    x0_data, x1_data, _ = inputs
+    x0 = chainer.Variable(x0_data)
+    x1 = chainer.Variable(x1_data)
+    loss = chainer_chemistry.functions.mean_squared_error(x0, x1,
+                                                          ignore_nan=True)
+    loss_value = cuda.to_cpu(loss.data)
+    assert loss.dtype == numpy.float32
+    assert loss_value.shape == ()
+
+    loss_expect = numpy.zeros(())
+    x0_data = cuda.to_cpu(x0_data)
+    x1_data = cuda.to_cpu(x1_data)
+    nan_mask = numpy.invert(numpy.isnan(x1_data)).astype(x1_data.dtype)
+    for i in numpy.ndindex(x0_data.shape):
+        loss_expect += ((x0_data[i] - numpy.nan_to_num(x1_data[i])) ** 2
+                        * nan_mask[i])
+    loss_expect /= x0_data.size
+    assert numpy.allclose(loss_value, loss_expect)
+
+
+def test_forward_cpu(inputs):
+    check_forward(inputs)
+    check_forward_ignore_nan(inputs)
+    check_forward_ignore_nan_with_nonnan_value(inputs)
+
+
+@pytest.mark.gpu
+def test_forward_gpu(inputs):
+    x0, x1, x2 = inputs
+    check_forward((cuda.to_gpu(x0), cuda.to_gpu(x1), None))
+    check_forward_ignore_nan((cuda.to_gpu(x0), None, cuda.to_gpu(x2)))
+
+
+def check_backward(inputs):
+    x0_data, x1_data, _ = inputs
+    gradient_check.check_backward(
+        chainer_chemistry.functions.mean_squared_error,
+        (x0_data, x1_data), None, eps=1e-2)
+
+
+def check_backward_ignore_nan(inputs):
+    x0_data, _, x2_data = inputs
+
+    def func(x0, x1):
+        return chainer_chemistry.functions.mean_squared_error(x0, x1,
+                                                              ignore_nan=True)
+    gradient_check.check_backward(func, (x0_data, x2_data), None, eps=1e-2)
+
```

Comment:
Could you add tests for backward and double backward with `ignore_nan=True` using `x0` and `x1`?

Reply:
fix